### ISSUE TITLE
IO::Path absolute is failed without path when Windows

### ISSUE THREAD

**Issue by ncaq** [rakudo/rakudo#4020]
## The Problem

`IO::Path` `absolute` is failed without `path` when Windows.

I use [.rc/commit-message-in-style at fa25d81390388487cea14a6c6106227288924e35 · ncaq/.rc](https://github.com/ncaq/.rc/blob/fa25d81390388487cea14a6c6106227288924e35/hooks/script/commit-message-in-style),
I found a problem.

ref <https://twitter.com/ncaq/status/1326130332186243073>
@lizmat 

## Expected Behavior

In all platforms, work.

## Actual Behavior

In Linux, work.
In Windows, fail.

## Steps to Reproduce

The minimal repro source code

```raku
my IO::Path $child = IO::Path.new(@*ARGS[0]);
$child.say;                     # "path.raku".IO
my IO::Path $root = IO::Path.new((run 'git', 'rev-parse', '--show-toplevel', :out).out.slurp(:close).trim);
$root.say;                       # "C:/Users/ncaq/Documents".IO
$root.path.say;                  # C:/Users/ncaq/Documents
$child.absolute($root.path).say; # C:\Users\ncaq\Documents\path.raku
# error!
# This type cannot unbox to a native string: P6opaque, IO::Path
# in block <unit> at path.raku
$child.absolute($root).say;
```

```console
% raku --ll-exception path.raku "path.raku"
"path.raku".IO
"C:/Users/ncaq/Documents".IO
C:/Users/ncaq/Documents
C:\Users\ncaq\Documents\path.raku
This type cannot unbox to a native string: P6opaque, IO::Path
   at SETTING::src/core.c/IO/Spec/Win32.pm6:82  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\runtime/CORE.c.setting.moarvm:is-absolute)
 from SETTING::src/core.c/IO/Spec/Win32.pm6:194  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\runtime/CORE.c.setting.moarvm:rel2abs)
 from SETTING::src/core.c/IO/Path.pm6:227  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\runtime/CORE.c.setting.moarvm:absolute)
 from SETTING::src/core.c/IO/Path.pm6:219  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\runtime/CORE.c.setting.moarvm:absolute)
 from path.raku:10  (<ephemeral file>:<unit>)
 from path.raku:1  (<ephemeral file>:<unit-outer>)
 from gen\moar\stage2\NQPHLL.nqp:1916  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\nqp\lib/NQPHLL.moarvm:eval)
 from gen\moar\stage2\NQPHLL.nqp:2121  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\nqp\lib/NQPHLL.moarvm:evalfiles)
 from gen\moar\stage2\NQPHLL.nqp:2081  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\nqp\lib/NQPHLL.moarvm:command_eval)
 from gen\moar\Compiler.nqp:55  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\lib/Perl6/Compiler.moarvm:command_eval)
 from gen\moar\stage2\NQPHLL.nqp:2006  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\nqp\lib/NQPHLL.moarvm:command_line)
 from gen\moar\rakudo.nqp:127  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\runtime\perl6.moarvm:MAIN)
 from gen\moar\rakudo.nqp:92  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\runtime\perl6.moarvm:<mainline>)
 from <unknown>:1  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\runtime\perl6.moarvm:<main>)
 from <unknown>:1  (C:\Users\ncaq\scoop\apps\rakudo-star\current\bin\..\share\perl6\runtime\perl6.moarvm:<entry>)
```

## Environment
* Operating system: `MINGW64_NT-10.0-18363 eagle 3.1.7-340.x86_64 2020-08-27 11:03 UTC x86_64 Msys`
* Compiler version (`perl6 -v` or `raku -v`): `This is Rakudo version 2020.05.1 built on MoarVM version 2020.05` (This version installed by scoop)

**Comment by lizmat** [comment]
Thanks for your report: I have now been able to reproduce the issue and am looking at fixing it.

**Comment by lizmat** [comment]
This turned out to be a missing coercion in the handling of paths on Windows.  Fixed with 08f5d028e3 , this will be in the 2020.11 release of Rakudo!  Thanks again for the report.  Please don't hesitate to report any other issues you may find on Windows: sadly we do not have enough eyes that look at Windows specific issues, so yours are very welcome!

**Comment by ncaq** [comment]
I was amazed at the quick fix.
It's great!
